src/wallet/wallet.cpp

```diff
@@ -4420,20 +4420,18 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
         walletInstance->m_min_fee = CFeeRate(n);
     }
 
-    walletInstance->m_allow_fallback_fee = Params().IsTestChain();
-    if (gArgs.IsArgSet("-fallbackfee")) {
-        CAmount nFeePerK = 0;
-        if (!ParseMoney(gArgs.GetArg("-fallbackfee", ""), nFeePerK)) {
-            chain.initError(strprintf(_("Invalid amount for -fallbackfee=<amount>: '%s'").translated, gArgs.GetArg("-fallbackfee", "")));
-            return nullptr;
-        }
-        if (nFeePerK > HIGH_TX_FEE_PER_KB) {
-            chain.initWarning(AmountHighWarn("-fallbackfee").translated + " " +
-                              _("This is the transaction fee you may pay when fee estimates are not available.").translated);
-        }
-        walletInstance->m_fallback_fee = CFeeRate(nFeePerK);
-        walletInstance->m_allow_fallback_fee = nFeePerK != 0; //disable fallback fee in case value was set to 0, enable if non-null value
+    CAmount nFeePerK = 0;
+    if (!ParseMoney(gArgs.GetArg("-fallbackfee", Params().DefaultFallbackfee()), nFeePerK)) {
+        chain.initError(strprintf(_("Invalid amount for -fallbackfee=<amount>: '%s'").translated, gArgs.GetArg("-fallbackfee", Params().DefaultFallbackfee())));
+        return nullptr;
     }
+    if (nFeePerK > HIGH_TX_FEE_PER_KB) {
+        chain.initWarning(AmountHighWarn("-fallbackfee").translated + " " +
+                          _("This is the transaction fee you may pay when fee estimates are not available.").translated);
+    }
+    walletInstance->m_fallback_fee = CFeeRate(nFeePerK);
+    walletInstance->m_allow_fallback_fee = nFeePerK != 0; //disable fallback fee in case value was set to 0, enable if non-null value
+
     if (gArgs.IsArgSet("-discardfee")) {
         CAmount nFeePerK = 0;
```

Comment:
This `nFeePerK` is shadowing the `nFeePerK` in the outer scope.

Reply:
Yes, it is. And more of the same below. But in every use of their own nFeePerK they initialize it.
Anyway, what do you think I should do?
Perhaps just rename the variable? any suggestion on the name?

Reply:
I would use another variable name to disambiguate :)

Reply:
Fixed nit, renamed to fee_per_k.